packages/machine/test/integration/set-state.spec.ts

```diff
@@ -0,0 +1,116 @@
+import NonceRegistry from "@counterfactual/contracts/build/NonceRegistry.json";
+import StateChannelTransaction from "@counterfactual/contracts/build/StateChannelTransaction.json";
+import { AssetType, NetworkContext } from "@counterfactual/types";
+import { WaffleLegacyOutput } from "ethereum-waffle";
+import { Contract, Wallet } from "ethers";
+import { AddressZero, WeiPerEther } from "ethers/constants";
+import { keccak256 } from "ethers/utils";
+
+import { SetStateCommitment } from "../../src/ethereum";
+import { StateChannel } from "../../src/models";
+
+import { toBeBNAndEq, toBeEq } from "./bignumber-jest-matcher";
+import { connectToGanache } from "./connect-ganache";
+import { getSortedRandomSigningKeys } from "./random-signing-keys";
+
+// To be honest, 30000 is an arbitrary large number that has never failed
+// to reach the done() call in the test case, not intelligently chosen
+const JEST_TEST_WAIT_TIME = 30000;
+
+// The AppRegistry.setState call _could_ be estimated but we haven't
+// written this test to do that yet
+const SETSTATE_COMMITMENT_GAS = 6e9;
+
+let networkId: number;
+let wallet: Wallet;
+let network: NetworkContext;
+let appRegistry: Contract;
+
+expect.extend({ toBeEq });
+expect.extend({ toBeBNAndEq });
+
+beforeAll(async () => {
+  [{}, wallet, networkId] = await connectToGanache();
+
+  const relevantArtifacts = [
+    { contractName: "AppRegistry", ...AppRegistry },
+    { contractName: "ETHBucket", ...ETHBucket },
+    { contractName: "NonceRegistry", ...NonceRegistry },
+    { contractName: "StateChannelTransaction", ...StateChannelTransaction }
+  ];
+
+  network = {
+    // Fetches the values from build artifacts of the contracts needed
+    // for this test and sets the ones we don't care about to 0x0
+    ETHBalanceRefund: AddressZero,
+    ...relevantArtifacts.reduce(
+      (accumulator: { [x: string]: string }, artifact: WaffleLegacyOutput) => ({
+        ...accumulator,
+        [artifact.contractName as string]: artifact.networks![networkId].address
+      }),
+      {}
+    )
+  } as NetworkContext;
+
+  appRegistry = new Contract(
+    (AppRegistry as WaffleLegacyOutput).networks![networkId].address,
+    AppRegistry.abi,
+    wallet
+  );
+});
+
+/**
+ * @summary Setup a StateChannel then set state on ETH Free Balance
+ */
+describe("Test", () => {
+  jest.setTimeout(JEST_TEST_WAIT_TIME);
+
+  it("test", async done => {
+    const signingKeys = getSortedRandomSigningKeys(2);
+
+    const users = signingKeys.map(x => x.address);
+
+    let stateChannel = StateChannel.setupChannel(network, AddressZero, users);
+    let freeBalanceETH = stateChannel.getFreeBalanceFor(AssetType.ETH);
+
+    const state = {
+      alice: stateChannel.multisigOwners[0],
+      bob: stateChannel.multisigOwners[1],
+      aliceBalance: WeiPerEther,
+      bobBalance: WeiPerEther
+    };
+
+    stateChannel = stateChannel.setState(freeBalanceETH.identityHash, state);
```

Comment:
You can use my new helper fn `setFreeBalanceFor(AssetType, AppState)`

Reply:
I wrote what I think is a better helper :p